src/overlays/actors/ovl_Boss_Tw/z_boss_tw.c

```diff
@@ -156,183 +187,5337 @@ static ColliderCylinderInit D_8094A840 = {
+            D_8094C876 = 10;
+            D_8094C874 = sp35;
+            D_8094C878 = 0;
+        }
+    }
+
+    if (msgId1 != 0) {
+        func_8010B680(globalCtx, msgId1, NULL);
+    }
+
+    switch (kotakeAnim) {
+        case 1:
+            Animation_MorphToLoop(&sKotakePtr->skelAnime, &D_0600230C, -5.0f);
+            break;
+        case 2:
+            Animation_MorphToLoop(&sKotakePtr->skelAnime, &D_06001D10, -5.0f);
+            break;
+        case 3:
+            Animation_MorphToLoop(&sKotakePtr->skelAnime, &D_060017E0, -5.0f);
+            break;
+        case 4:
+            Animation_MorphToLoop(&sKotakePtr->skelAnime, &D_060012A4, -5.0f);
+            break;
+    }
+
+    switch (koumeAnim) {
+        case 1:
+            Animation_MorphToLoop(&sKoumePtr->skelAnime, &D_0600230C, -5.0f);
+            break;
+        case 2:
+            Animation_MorphToLoop(&sKoumePtr->skelAnime, &D_06001D10, -5.0f);
+            break;
+        case 3:
+            Animation_MorphToLoop(&sKoumePtr->skelAnime, &D_060017E0, -5.0f);
+            break;
+        case 4:
+            Animation_MorphToLoop(&sKoumePtr->skelAnime, &D_060012A4, -5.0f);
+            break;
+    }
+
+    if (this->work[CS_TIMER_2] >= 120 && this->work[CS_TIMER_2] < 500) {
+        Math_ApproachF(&this->workf[UNK_F18], 255.0f, 0.1f, 5.0f);
+    }
+
+    if (this->work[CS_TIMER_2] >= 150) {
+        Math_ApproachF(&sKoumePtr->workf[UNK_F17], (Math_SinS(this->work[CS_TIMER_1] * 2000) * 0.05f) + 0.4f, 0.1f,
+                       0.01f);
+        Math_ApproachF(&sKotakePtr->workf[UNK_F17], (Math_CosS(this->work[CS_TIMER_1] * 1700) * 0.05f) + 0.4f, 0.1f,
+                       0.01f);
+
+        if (this->work[CS_TIMER_2] >= 880) {
+            Math_ApproachF(&sKotakePtr->actor.world.pos.y, 2000.0f, 1.0f, this->actor.speedXZ);
+            Math_ApproachF(&sKoumePtr->actor.world.pos.y, 2000.0f, 1.0f, this->actor.speedXZ);
+            Math_ApproachF(&this->actor.speedXZ, 10.0f, 1.0f, 0.25f);
+
+            if (this->work[CS_TIMER_2] >= 930) {
+                Math_ApproachF(&this->workf[UNK_F19], 5.0f, 1.0f, 0.05f);
+                Math_ApproachF(&this->workf[UNK_F18], 0.0f, 1.0f, 3.0f);
+            }
+
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_GOTO_HEAVEN - SFX_FLAG);
+        } else {
+            f32 yTarget = Math_CosS(this->work[CS_TIMER_2] * 1700) * 4.0f;
+            Math_ApproachF(&sKotakePtr->actor.world.pos.y, 20.0f + (263.0f + yTarget), 0.1f, this->actor.speedXZ);
+            yTarget = Math_SinS(this->work[CS_TIMER_2] * 1500) * 4.0f;
+            Math_ApproachF(&sKoumePtr->actor.world.pos.y, 20.0f + (263.0f + yTarget), 0.1f, this->actor.speedXZ);
+            Math_ApproachF(&this->actor.speedXZ, 1.0f, 1.0f, 0.05f);
+        }
+    }
+}
+
+#ifdef NON_MATCHING
+void BossTw_TwinrovaDeathCS(BossTw* this, GlobalContext* globalCtx) {
+    s16 i;
+    Vec3f spD8;
+    Player* player = PLAYER;
+    Camera* mainCam = Gameplay_GetCamera(globalCtx, 0);
```

Comment:
```suggestion
    Camera* mainCam = Gameplay_GetCamera(globalCtx, MAIN_CAM);
```
same with other `Gameplay_GetCamera` calls